src/unstract/sdk/prompt.py

```diff
@@ -63,17 +68,28 @@ def _post_call(
             "structure_output": "",
         }
         url: str = f"{self.base_url}/{url_path}"
-        headers: dict[str, str] = {
-            "Authorization": f"Bearer {self.bearer_token}"
-        }
+        headers: dict[str, str] = {"Authorization": f"Bearer {self.bearer_token}"}
+        response: Response = Response()
```

Comment:
NIT: This seems valid since we've explicitly handled `Timeout` and `ConnectionError` but please keep in mind that if the `response` is not actually returned in line 75 and there's a `RequestException` while calling `requests.post()` we might end up making the below calls which could result in an erroneous behaviour
```
response.headers.get("Content-Type", "").lower()
response.json()
response.text
```
I'm not sure if these might return anything valid for the default constructed `Response` object

Reply:
@chandrasekharan-zipstack 
1. We have handled `Timeout` and `ConnectionError` - which does not have the response variable. So this doesn't go into RequestException
2. If RequestException doesnt return any Response object - [This validation](https://github.com/Zipstack/unstract-sdk/blob/b2bb0786d28ea500f6fba5bceebcbf0626d74662/src/unstract/sdk/prompt.py#L101) takes care and the stringified error message will be returned.